new errors on build in vs2010

##### System information (version)
- OpenCV => 3.2
- Operating System / Platform = Windows 32 Bit
- Compiler = Visual Studio 2010

##### Detailed description
Hi, I was trying to build opencv with xfeatures2d module while using Visual Studio 2010 and i got the following errors:
```
41>..\..\..\..\opencv_contrib\modules\xfeatures2d\src\pct_signatures.cpp(400): error C2668: 'cv::sqrt' : ambiguous call to overloaded function
41>          C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\include\math.h(589): could be 'long double sqrt(long double)'
41>          C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\include\math.h(541): or       'float sqrt(float)'
41>          C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\include\math.h(127): or       'double sqrt(double)'
41>          while trying to match the argument list '(const int)'

41>..\..\..\..\opencv_contrib\modules\xfeatures2d\src\harris_lapace_detector.cpp(107): error C2668: 'log' : ambiguous call to overloaded function
41>          C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\include\math.h(575): could be 'long double log(long double)'
41>          C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\include\math.h(527): or       'float log(float)'
41>          C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\include\math.h(120): or       'double log(double)'

41>..\..\..\..\opencv_contrib\modules\xfeatures2d\src\boostdesc.cpp(235): error C3861: 'round': identifier not found
```


I think these errors are caused by the use of an older C++ standard(C++98).

Does the opencv_contrib repository continues to follow the OpenCV Coding Style Guide (http://code.opencv.org/projects/opencv/wiki/Coding_Style_Guide)?

> ie: Formally, the code must comply with the C++ 98 standard

If so maybe this is an issue.

Also while I was searching for an already posted issue into this matter I noticed (https://github.com/opencv/opencv_contrib/issues/754) that deals with problems with builds with visual studio 2010. So i was wondering, is there an automatic build system for opencv that deals with this checks?

Thank you all in advance!

--- a/modules/xfeatures2d/src/boostdesc.cpp
+++ b/modules/xfeatures2d/src/boostdesc.cpp
@@ -232,7 +232,7 @@ static void computeGradientMaps( const Mat& im,
               break;
 
             case ASSIGN_HARD_MAGN:
-              gradMap[index].at<uchar>(i,j) = (uchar) round( gradMagnitude );
+              gradMap[index].at<uchar>(i,j) = (uchar) cvRound( gradMagnitude );
               break;
 
             case ASSIGN_BILINEAR:
@@ -240,8 +240,8 @@ static void computeGradientMaps( const Mat& im,
               index2 = ( index2 == orientQuant ) ? 0 : index2;
               binCenter  = ( index + 0.5f ) * binSize;
               weight = 1 - abs( theta - binCenter ) / binSize;
-              gradMap[index ].at<uchar>(i,j) = (uchar) round( 255 * weight );
-              gradMap[index2].at<uchar>(i,j) = (uchar) round( 255 * ( 1 - weight ) );
+              gradMap[index ].at<uchar>(i,j) = (uchar) cvRound( 255 * weight );
+              gradMap[index2].at<uchar>(i,j) = (uchar) cvRound( 255 * ( 1 - weight ) );
               break;
 
             case ASSIGN_SOFT:
@@ -251,7 +251,7 @@ static void computeGradientMaps( const Mat& im,
                 binCenter = ( index2 + 0.5f ) * binSize;
                 weight = cos( theta - binCenter );
                 weight = ( weight < 0 ) ? 0 : weight;
-                gradMap[index2].at<uchar>(i,j) = (uchar) round( 255 * weight );
+                gradMap[index2].at<uchar>(i,j) = (uchar) cvRound( 255 * weight );
               }
               break;
 
@@ -262,7 +262,7 @@ static void computeGradientMaps( const Mat& im,
                 binCenter = ( index2 + 0.5f ) * binSize;
                 weight = cos( theta - binCenter );
                 weight = ( weight < 0 ) ? 0 : weight;
-                gradMap[index2].at<uchar>(i,j) = (uchar) round( gradMagnitude * weight );
+                gradMap[index2].at<uchar>(i,j) = (uchar) cvRound( gradMagnitude * weight );
               }
               break;
           } // end switch

--- a/modules/xfeatures2d/src/harris_lapace_detector.cpp
+++ b/modules/xfeatures2d/src/harris_lapace_detector.cpp
@@ -104,7 +104,7 @@ void Pyramid::build(const Mat& img, bool DOG)
 
     Size imgSize = img.size();
     int minSize = MIN(imgSize.width, imgSize.height);
-    int octavesN = MIN(params.octavesN, int(floor(log((double) minSize)/log(2))));
+    int octavesN = MIN(params.octavesN, int(floor(log((double) minSize)/log((float)2))));
     float sigma0 = params.sigma0;
     float sigma = sigma0;
     int layersN = params.layersN + 3;

--- a/modules/xfeatures2d/src/pct_signatures.cpp
+++ b/modules/xfeatures2d/src/pct_signatures.cpp
@@ -397,7 +397,7 @@ namespace cv
                 break;
             case REGULAR:
             {
-                int gridSize = (int)ceil(sqrt(count));
+                int gridSize = (int)ceil(sqrt((float)count));
                 const float step = 1.0f / gridSize;
                 const float halfStep = step / 2;
                 float x = halfStep;

